### PR TITLE
Add two new fields for the member feature

### PR DESCRIPTION
**Pull request by kangmingtay**
- [x] Add total points field to accumulate all the points the member has earned over time.

- [x] Add email to the member details

### CODE REVIEW

File: src/seedu/addressbook/data/member/Points.java [hunk 1]
```diff
@@ -12,14 +12,22 @@
...
     public Points() {
-        this.value = 0;
+        this.currentPoints = 0;
+        this.totalPoints = 0;
     }
 
-    public Points(int points) {
-        this.value = points;
+    public Points (int pointsToRedeem) {
+        this.currentPoints = pointsToRedeem;
```
Comment (px1099) [inline src/seedu/addressbook/data/member/Points.java:26]: Need to assign `totalPoint = pointsToRedeem`

File: src/seedu/addressbook/data/member/Member.java [hunk 2]
```diff
@@ -55,13 +55,17 @@ public MemberName getName() {
...
         return points;
     }
 
     @Override
     public int getPointsValue() {
-        return points.getPoints();
+        return points.getCurrentPoints();
+    }
+
+    public int getTotalPointsValue() {
```
Comment (px1099) [inline src/seedu/addressbook/data/member/Member.java:78]: Add this to `ReadOnlyMember` and add `@override` here

File: src/seedu/addressbook/data/order/Order.java [hunk 3]
```diff
@@ -129,12 +129,16 @@ public int getMaxPointsRedeemable() {
         return Integer.min(pointsLimitByPrice, pointsLimitByMember);
     }
 
+    public int getEarnedPointsValue(double price) {
```
Comment (px1099) [inline src/seedu/addressbook/data/order/Order.java]: add `@override`

Comment (px1099) [reply src/seedu/addressbook/data/order/Order.java]: Price is available in `Order.java`, there is no need to pass it in

**Review by px1099 — CHANGES_REQUESTED** [review]

File: src/seedu/addressbook/data/member/Points.java (hunk 1 above)
Comment (kangmingtay) [reply src/seedu/addressbook/data/member/Points.java:26]: is this to avoid null pointer exception?

**Comment by px1099** [comment]
#194 
Bug found in `getMaxPointRedeemable()` in `Order.java`:
`int pointsLimitByPrice = Points.getEarnedPointsValue(getOriginalPrice());` is wrong
Should have been:
`int pointsLimitByPrice = Points.getRedeemedPointsValue(getOriginalPrice());`
This is because we are trying to do reverse calculation to see how many point can we use

File: src/seedu/addressbook/data/member/Points.java (hunk 1 above)
Comment (px1099) [reply src/seedu/addressbook/data/member/Points.java:26]: Yes

**Review by px1099 — APPROVED** [review]

**Pull request merged**